sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -526,7 +526,7 @@ public static ResourceIdentifier Parse(string input)
         /// If the method returns false, result will be null.
         /// </param>
         /// <returns> True if the parse operation was successful; otherwise, false. </returns>
-        public static bool TryParse(string input, out ResourceIdentifier? result)
+        public static bool TryParse(string? input, out ResourceIdentifier? result)
```

Comment:
need `[MaybeNullWhen(false)]` on the result parameter.

Line 535 needs a `!` on the input parameter since we do the null check on 532 this is safe.

Reply:
@christothes do you know who owns our nullability store in azure.core.  Is this a back compat change?

Reply:
thanks @m-nash for the suggestions, I've applied them, and the build is now green

Reply:
Yes, adding nullability shouldn't be a breaking change.